### DISCUSSION TITLE
Django Application not serving media files

### DISCUSSION THREAD

**Discussion by rasulkireev** [caprover/caprover#1779] [General]
My django application is saving file locally in the `media` directory via the following settings:
```
MEDIA_URL = "/media/"
MEDIA_ROOT = os.path.join(BASE_DIR, "media/")
```

In my caprover instance I create a persistent app and create the following mapping:
![image](https://github.com/caprover/caprover/assets/8257474/6db9fe39-4bec-44c8-8da7-155c02263920)

I confirmed that files exists both on the server and in the container:

container
<img width="630" alt="image" src="https://github.com/caprover/caprover/assets/8257474/11aaebbe-5ece-4570-bad1-db95be83fc78">

server
<img width="653" alt="image" src="https://github.com/caprover/caprover/assets/8257474/730f09d9-5a66-4170-b830-9d89a9c4a51c">

Finally, I need to "serve" the files via NGINX if I'm not mistaken. I have tried adding the following blocks to my NGINX config in "HTTP Settings":

```
location /media/ {
    alias /app/media/;
}
```

```
location /media/ {
    alias /app/media;
}
```

```
location /media/ {
    alias /app/;
}
```

```
location /media/ {
    alias /app;
}
```

And finally all the above but with `root` instead of `alias`.

I understand that this might be more of a NGINX question as opposed to Caprover question, but would love it if someone spots something that I'm doing wrong.

P.S. I saw @githubsaturn mentioned a "second nginx config" a few times. Does he mean the one that I can edit in the Caprover UI?

P.P.S. Some related Issues that did not help me:
https://github.com/caprover/caprover/issues/1709
https://github.com/caprover/caprover/issues/225
https://github.com/caprover/caprover/issues/888
https://github.com/caprover/caprover/issues/938

**Comment by githubsaturn** [comment]
> I need to "serve" the files via NGINX if I'm not mistaken

No. You don't have to. Serving the content directly from nginx is a lot more complex and it adds very little benefit to your application's performance. 

So, I suggest just directly serving these files from the Django app.
